extension/src/resources.ts

```diff
@@ -40,28 +38,27 @@ function readJsonDataFile(datafile: string, intoObject: IJsonLoadable): void {
 }	
 
 
-function readDataFileByLine(datafile: string, action: (items: string[]) => void) {
-	var fs = require("fs");
+function readDataFileByLine(datafile: string,action: (items: string[]) => void) {
 	var dataPath = path.resolve(__dirname, datafile);
-	fs.readFile(dataPath, "utf8", function(err: Error, data: string) {
-		if (err === null) {
-			if (data.includes("\r\n")) {
-				action(data.split("\r\n"));
-			}
-			else {
-				action(data.split("\n"));
-			}
+	try {
+		var data = fs.readFileSync(dataPath,{encoding:'utf8', flag:'r'});
+		if (data.includes("\r\n")) {
+			action(data.split("\r\n"));
 		}
-	});
+		else {
+			action(data.split("\n"));
+		}
+	} catch (error) {
+		console.log(error);
+	}
 }
 
 
 function readDataFileByDelimiter(datafile: string, delimiter: string, action: (item: string) => void) {
-	var fs = require("fs");
 	var dataPath = path.resolve(__dirname, datafile);
-	fs.readFile(dataPath, "utf8", function(err: Error, data: string) {
+	try {
+		var data = fs.readFileSync(dataPath,{encoding:'utf8', flag:'r'});
 		var lineList = new Array<String>();
```

Comment:
I think the code scanning is correct, there's no need to initialize lineList

Reply:
yes, I have changed it to var lineList : Array<String>